sources/packages/backend/apps/api/src/route-controllers/application-exception/_tests_/e2e/application-exception.aest.approveException.e2e-spec.ts

```diff
@@ -0,0 +1,221 @@
+    // Arrange
+    const updateApplicationException = {
+      exceptionStatus: ApplicationExceptionStatus.Approved,
+      noteDescription: faker.lorem.text(10),
+    };
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .patch("/aest/application-exception/9999999")
+      .send(updateApplicationException)
+      .auth(
+        await getAESTToken(AESTGroups.BusinessAdministrators),
+        BEARER_AUTH_TYPE,
+      )
+      .expect(HttpStatus.NOT_FOUND)
+      .expect({
+        statusCode: 404,
+        message: "Student application exception not found.",
+        error: "Not Found",
+      });
+  });
+
+  it("Should not be able to approve application exception when it is not in pending status.", async () => {
+    // Arrange
+    const application = await createFakeApplicationWithApplicationException(
+      ApplicationExceptionStatus.Declined,
+      appDataSource,
+      appModule,
+    );
+    const updateApplicationException = {
+      exceptionStatus: ApplicationExceptionStatus.Approved,
+      noteDescription: faker.lorem.text(10),
+    };
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .patch(
+        `/aest/application-exception/${application.applicationException.id}`,
+      )
+      .send(updateApplicationException)
+      .auth(
+        await getAESTToken(AESTGroups.BusinessAdministrators),
+        BEARER_AUTH_TYPE,
+      )
+      .expect(HttpStatus.UNPROCESSABLE_ENTITY)
+      .expect({
+        statusCode: 422,
+        message:
+          "Student application exception must be in Pending state to be assessed.",
+        error: "Unprocessable Entity",
+      });
+  });
+
+  it("Should not be able to approve application exception when status passed is not a valid application exception status.", async () => {
+    // Arrange
+    const application = await createFakeApplicationWithApplicationException(
+      ApplicationExceptionStatus.Declined,
+      appDataSource,
+      appModule,
+    );
+    const updateApplicationException = {
+      exceptionStatus: ApplicationExceptionStatus.Pending,
+      noteDescription: faker.lorem.text(10),
+    };
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .patch(
+        `/aest/application-exception/${application.applicationException.id}`,
+      )
+      .send(updateApplicationException)
+      .auth(
+        await getAESTToken(AESTGroups.BusinessAdministrators),
+        BEARER_AUTH_TYPE,
+      )
+      .expect(HttpStatus.BAD_REQUEST)
+      .expect({
+        statusCode: 400,
+        message: [
+          "exceptionStatus must be one of the following values: Approved, Declined",
+        ],
+        error: "Bad Request",
+      });
+  });
+
+  it("Should not be able to approve application exception when note is empty.", async () => {
+    // Arrange
+    const application = await createFakeApplicationWithApplicationException(
```

Comment:
When the exception is thrown before reaching the controller body we can avoid creating DB data.